Header   header
int32    intField
float64  doubleField
string   stringField
bool     boolField